consensus/hotstuff/tracker/tracker.go

```diff
@@ -16,6 +16,33 @@ type NewestQCTracker struct {
 	newestQC *atomic.UnsafePointer
 }
 
+type EntityCompareFunc[E any] func(a, b E) bool
```

Comment:
we would need a convention that `nil` is always older (principle of [vacuous truth](https://en.wikipedia.org/wiki/Vacuous_truth))